manifests/init.pp

```diff
@@ -14,6 +14,13 @@
         require => Exec["update package lists for nginx"],
     }
 
+    file { "nginx log dir":
+        path => "/var/log/nginx",
+        ensure => "directory",
+        owner => "www-data",
```

Comment:
Curious: This smells like a packaging issue with the NGINX version we install or are we using a non-default logging location (which it doesn't seem like)?

Reply:
I had a very quick look at this on a fresh bionic machine, the only difference between the install there and the one on meter-points-service in oegb-prod seems to be this:

```
ssm-user@ip-10-0-29-16:~$ sudo ls -ld /var/log/nginx/
drwxr-x--- 2 root adm 4096 Feb  2 17:42 /var/log/nginx/
```

```
ubuntu@corking-trumpetfish:~$ ls -ld /var/log/nginx/
drwxr-xr-x 2 root adm 4096 Feb  3 00:07 /var/log/nginx/
```

Reply:
I also expect the webserver to run as root and not www-data. The issue sounds more like we aren't sending the SIGHUP to nginx when rotating the files.

Reply:
I did some testing yesterday, and wasn't able to rotate the files unless either the user running the worker process owns the directory, or others can write to it.

```
postrotate
		PID=$(awk 'BEGIN { RS="[;{}]" } { if ($1 == "pid") { print $2; exit; } }' /etc/nginx/nginx.conf)
		start-stop-daemon --stop --signal USR1 --quiet --pidfile ${PID:-/run/nginx.pid} --name nginx
endscript
```

```
postrotate
		invoke-rc.d nginx -s reopen >/dev/null 2>&1
endscript
```

```
postrotate
		kill -USR1 `cat /run/nginx.pid`
endscript
```

Neither of the above worked. What did work leaving the logrotate as it is:

```
postrotate
		invoke-rc.d nginx rotate >/dev/null 2>&1
endscript
```
and changing the owner to `www-data` or `0751` mode. I'll do some more testing today

Reply:
Right, the only reason is others having no permissions on the directory. Out of the box permissions to `/var/log/nginx` are `0755` and on our EC2 instances I can observe `0750`. If this is getting changed after the deployment, then changing it here won't make a difference. I'll investigate some more

Reply:
After the conversation here: https://octoenergy.slack.com/archives/C01FM6D03PX/p1648810822875839 the initial change is indeed the change to make to the NGINX module, to keep CIS hardening: `4.2.4 Ensure permissions on all logfiles are configured` rule intact.

So the change will make `www-data` the owner of `/var/log/nginx` and set the permissions to `0755`. If CIS hardening cron job kicks in - that's fine, rotation will continue to work, here's an example on an `api-server` instance on `oegb-test`:

![Screenshot 2022-04-01 at 16 58 08](https://user-images.githubusercontent.com/60039710/161301032-f992f758-6ef3-4c14-80a9-2d71981b4f6e.png)

I made some requests to make sure it works for `access.log` too just to make sure:

![Screenshot 2022-04-01 at 17 06 17](https://user-images.githubusercontent.com/60039710/161301084-176afc05-4303-4f87-803d-85ea208f7615.png)